### PR TITLE
Reset password emailing

### PR DESCRIPTION
**Pull request by ad08fee3**
Pull it and make sure it works!

### CODE REVIEW

File: dli_app/mod_auth/controllers.py [hunk 1]
```diff
@@ -103,3 +115,67 @@ def logout():
...
+        # Email is authenticated and sent
+        email = form.email.data
+	#Create User Specific URL here
+        pw_reset = PasswordReset(
+            user= User.get_by_email(email),
+            key=''.join(
+                random.choice(
+                    string.ascii_letters + string.digits
+                ) for _ in range(60)
+            ),
```
Comment (gorel) [inline dli_app/mod_auth/controllers.py:138]: Personal suggestion:
Change PasswordReset to set key=None as default, and within **init**, if key=None, generate it there.
That will make this section a bit less cluttered.

``` python
def __init__(self, user, key=None):
    if key is None:
        key = random.choice(...etc...)
    self.user = user
    self.key = key
```

File: dli_app/mod_auth/controllers.py [hunk 2]
```diff
@@ -11,6 +13,7 @@
     render_template,
     request,
     url_for,
+    current_app
```
Comment (gorel) [inline dli_app/mod_auth/controllers.py:16]: [nit]
Final comma after current_app for consistency

File: dli_app/mod_auth/controllers.py [hunk 3]
```diff
@@ -103,3 +115,67 @@ def logout():
...
+            user= User.get_by_email(email),
+            key=''.join(
+                random.choice(
+                    string.ascii_letters + string.digits
+                ) for _ in range(60)
+            ),
+        )
+        db.session.add(pw_reset)
+        db.session.commit()
+        #ADD EMAIL STUFF HERE
```
Comment (gorel) [inline dli_app/mod_auth/controllers.py:142]: Remove this comment, or make it better in some way.

File: dli_app/mod_auth/controllers.py [hunk 4]
```diff
@@ -103,3 +115,67 @@ def logout():
...
+	mail=Mail(current_app)
+	title='Reset your Password'
+        url="http://68.234.146.84:PORT /auth/setnewpass/"+pw_reset.key
+        content='Click this link to reset your password: '+url
+	sender='cs490testing@gmail.com'
+	msg=Message(title,sender=sender,recipients=[email])
+	msg.body=content
+	mail.send(msg)
```
Comment (gorel) [inline dli_app/mod_auth/controllers.py:150]: This section has a lot of inconsistent spacing.
Also, only use var=value in function calls. In general, you should put a single space around operators

``` python
mail = Mail(current_app)
title = 'Reset your Password'
etc.
```

File: dli_app/mod_auth/models.py [hunk 5]
```diff
@@ -148,6 +152,29 @@ def get_by_email(cls, email):
...
+    key = db.Column(db.String(64))
+
+    def __init__(self, user, key):
+        """Initialize a  model"""
+        self.user = user
+	self.key = key
+    def __repr__(self):
+	"""Return a descriptive representation of password reset"""
+	return '<Reset password for use %r>' % self.user
+    @classmethod
```
Comment (gorel) [inline dli_app/mod_auth/models.py:168]: Newline before method

File: dli_app/mod_auth/models.py [hunk 6]
```diff
@@ -148,6 +152,29 @@ def get_by_email(cls, email):
...
+    user_id= db.Column(db.Integer, db.ForeignKey('user.id'))
+    key = db.Column(db.String(64))
+
+    def __init__(self, user, key):
+        """Initialize a  model"""
+        self.user = user
+	self.key = key
+    def __repr__(self):
+	"""Return a descriptive representation of password reset"""
+	return '<Reset password for use %r>' % self.user
```
Comment (gorel) [inline dli_app/mod_auth/models.py:167]: [nit]
s/for use/for user/

File: dli_app/mod_reports/models.py [hunk 7]
```diff
@@ -424,9 +424,10 @@ def finalize(self):
...
 
 class FieldTypeConstants():
-    """Constant FieldTypes used for easy type-checking in other modules"""
+    """Constant FieldTypes used for easy type-checking in other modules
     CURRENCY = FieldType.query.filter_by(name="currency").first()
     DOUBLE = FieldType.query.filter_by(name="double").first()
     INTEGER = FieldType.query.filter_by(name="integer").first()
     STRING = FieldType.query.filter_by(name="string").first()
-    TIME = FieldType.query.filter_by(name="time").first()
+    TIME = FieldType.query.filter_by(name="time").first()"""
```
Comment (gorel) [inline dli_app/mod_reports/models.py:432]: All of this needs uncommented

File: dli_app/templates/auth/login.html [hunk 8]
```diff
@@ -28,5 +28,6 @@
         </div>
       </div>
     </form>
+    <a href="http://68.234.146.84:8222/auth/resetpass/">I forgot my password</a>
```
Comment (gorel) [inline dli_app/templates/auth/login.html:31]: Don't hardcode this. This is where jinja2 shines.

``` python
<a href="{{ url_for('auth.resetpass') }}">
```

That way if we ever rename the resetpass method, our templates will automatically update.

File: dli_app/templates/auth/resetpass.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{% extends 'layout.html' %}
+{% block body %}
+<div class="page-header">
+    <h1>Reset Password</h1>
+</div>
+
+<div class="col-md-6">
+    <form method="POST" action="{{ url_for('auth.resetpass', next='/') }}" class="form-horizontal">
```
Comment (gorel) [inline dli_app/templates/auth/resetpass.html:8]: You don't need the next parameter. Your resetpass method redirects to home anyways.

**Comment by gorel** [comment]
Looks pretty good! I have a few change requests, but nothing major.
Except commenting out all of FieldTypeConstants is certain to break some things later. lol

**Comment by gorel** [comment]
Evan: Finish this up by this weekend?

**Comment by ad08fee3** [comment]
Continued in https://github.com/gorel/dli-reports/pull/65 because rebasing was being silly. Or I may just be bad at it or lazy.

**Pull request closed without merging**